quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/commons/CodeActionFactory.java

```diff
@@ -56,9 +57,24 @@ public static CodeAction remove(String title, Range range, TextDocumentItem docu
 	 */
 	public static CodeAction insert(String title, Position position, String insertText, TextDocumentItem document,
 			Diagnostic diagnostic) {
+		return insert(title, position, insertText, document, Arrays.asList(diagnostic));
+	}
+
+	/**
+	 * Create a CodeAction to insert a new content at the end of the given range.
+	 * 
+	 * @param title
+	 * @param range
+	 * @param insertText
+	 * @param document
+	 * @param diagnostics
+	 * @return
+	 */
+	public static CodeAction insert(String title, Position position, String insertText, TextDocumentItem document,
+			List<Diagnostic> diagnostics) {

```

Comment:
Are you sure to update this method since now you generate one codeaction?

Reply:
Do you wish I remove 

` insert(String title, Position position, String insertText, TextDocumentItem document, Diagnostic diagnostic)`

since its not being used anymore?

The reason why I'm using
`insert(String title, Position position, String insertText, TextDocumentItem document, List<Diagnostic> diagnostics)`
is because I am providing a list of all required properties diagnostics for the CodeAction to insert all missing required properties

Reply:
Ok I understand, please keep the insert signature with one Diagnostic. This CodeActionFactory (copied from LSP4XML) could be used for any LSP language server (commons classes).